git_md_page/views.py

```diff
@@ -0,0 +1,38 @@
+"""Views for git_md_page."""
+import hmac
+import json
+
+from django.http import HttpResponse, HttpResponseForbidden, HttpResponseNotFound
+from django.utils.crypto import constant_time_compare
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_POST
+
+from git_md_page.models import GitRepository, GitTextPluginModel
+from git_md_page.signals.git_update import repo_update
+
+
+@csrf_exempt
+@require_POST
+def git_update_endpoint(request, pk):
+    """Validate incoming message and trigger update."""
+    # First, get the repository
+    try:
+        repo = GitRepository.objects.get(pk=pk)
+    except GitRepository.DoesNotExist:
+        return HttpResponseNotFound("Repository does not exist.")
+
+    # Get the header and verify signature
+    sig_header = request.META.get("HTTP_X_HUB_SIGNATURE")
+    if sig_header is None:
+        return HttpResponseForbidden("No signature")
+    signature = "sha1=" + hmac.new(repo.secret.encode(), request.body, digestmod="sha1").hexdigest()
+    if not constant_time_compare(signature, sig_header):
+        return HttpResponseForbidden("Access denied")
+
+    # Now parse the body for event and check whether or not is the commit on master
+    parsed = json.loads(request.body.decode())
+    if parsed.get("ref") != "refs/heads/{}".format(repo.branch):
+        return HttpResponse()
```

Comment:
What's that line supposed to do exactly? No redirect, no error message, nothing?

Reply:
Honestly, no idea. I think that the response actually is not even relevant, only the status code might be (for example for bugs/failure reports), but I really don't know how exactly does it work...

Reply:
The response is useful for [debugging](https://docs.github.com/en/developers/webhooks-and-events/testing-webhooks#response). It would be better to return `HttpResponse("Branch ignored")`.

Reply:
Fixed. Also the `happy` response changed to `success`.